Dockerfile

```diff
@@ -0,0 +1,24 @@
+# Builder stage
+FROM node:16-alpine3.13 as builder
+# Install dependencies for building node modules
+# python3, g++, make: required by node-gyp
+# git: required by check-update-github
+WORKDIR /app
+COPY ./package.json ./package.json
+RUN apk add --no-cache --virtual \
+    .gyp \
+    python3=~3.8 \
+    make=~4.3 \
+    g++=~10.2 \
+    git=~2.30 \
+    && npm install --only=production \
+    && apk del .gyp
+
+# App stage
+FROM node:16-alpine3.13 as app
```

Comment:
Can this build omit changes to the build environment to reduce the size of the final packaged image?

Reply:
I'm not sure what you meant, can you propose an example or any link to the docs? I will read & optimize it.

From what I understand, currently Dockerfile is using multi-stage build, the build stage does not affect the size of the final image.

Reply:
I've figured it out, I didn't know about such multi-stage build dockerfile before.
I rewrote the dockerfile I made before.
The final image is much smaller, thanks

Reply:
Great to know 😄 .
I don't know any other way to make it smaller. It's already very small to download. I will mark the conversation as resolved 🎉 .